src/readers/LiquidationReader.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: BUSL-1.1
+// This code is made available under the terms and conditions of the Business Source License 1.1 (BUSL-1.1).
+// The act of publishing this code is driven by the aim to promote transparency and facilitate its utilization for educational purposes.
+
+pragma solidity 0.8.18;
+
+// interfaces
+import { IPerpStorage } from "@hmx/storages/interfaces/IPerpStorage.sol";
+import { ICalculator } from "@hmx/contracts/interfaces/ICalculator.sol";
+
+// libs
+import { SqrtX96Codec } from "@hmx/libraries/SqrtX96Codec.sol";
+import { TickMath } from "@hmx/libraries/TickMath.sol";
+
+contract LiquidationReader {
+  IPerpStorage immutable perpStorage;
+  ICalculator immutable calculator;
+
+  error LiquidationReader_InvalidArray();
+
+  constructor(address _perpStorage, address _calculator) {
+    perpStorage = IPerpStorage(_perpStorage);
+    calculator = ICalculator(_calculator);
+  }
+
+  /// @notice Get the liquidatable sub-accounts.
+  /// @param _limit The maximum number of sub-accounts to retrieve.
+  /// @param _offset The offset for fetching sub-accounts.
+  /// @param _assetIds An array of asset IDs.
+  /// @param _pricesE8 An array of prices in E8 format corresponding to the asset IDs.
+  /// @param _shouldInverts An array of boolean values indicating whether to invert prices for the asset IDs.
+  /// @return liquidatableSubAccounts An array of sub-account addresses that meet the liquidation criteria.
+  function getLiquidatableSubAccount(
+    uint64 _limit,
+    uint64 _offset,
```

Comment:
Limit, offset is not used as pagination of "liquidatable subaccount", but "active sub account" instead. Not sure if this is an intention.